openapi-format.js

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 "use strict";
 
-const traverse = require('traverse');
+const traverse = require('neotraverse');
```

Comment:
Should be `neotraverse/legacy` 😄

Reply:
Why? engines in package.json says Node >=18 🤔

Reply:
neotraverse doesn't ship with CommonJS. That's for /legacy build 😄

Reply:
The test suite is failing `Cannot find module 'neotraverse/legacy' from 'openapi-format.js'`
More details: 
https://github.com/thim81/openapi-format/actions/runs/9987096142/job/27614138151?pr=117

Reply:
I found out the issue, it's to do with how `this` in traverse methods is used and accessed. (`get(this, 'parent.parent.key')` etc). This matters because in neotraverse, internally we're using a class, whereas traverse was a function constructor with methods attached to the prototype. 

This will require refactoring some code slightly, which is up to @wojtekmaj. If he's fine with refactoring, all the better, but in case not, I'd still suggest that you pin 0.6.8 of traverse explicitly and never upgrade it(unless you're refactoring this package into ESM, in whic case neotraverse will shine 😄). That alone is enough to reduce bloat size a lot!

Reply:
hi @PuruVJ,

I already appreciate that you and @wojtekmaj have taken the time to create this PR.
I'm a bit confused about the test failing because ` Cannot find module 'neotraverse/legacy' from 'openapi-format.js'` and the `this` function <> Class.

Looking at the Neotraverse docs, `this.parent.parent.key`seems to be supported, so if understand better what is wrong I could refactor it myself.

The `get(this, 'parent.parent.key')` is just little helper utility, since before NodeJS18 optional chaining was not possible, so this [get](https://github.com/thim81/openapi-format/blob/ccbd2c1d615ed51ce3c58f1a9bd6b41e0bac28c5/utils/filtering.js#L40) was used to not riddle the code with more IF statements.

Reply:
So, a few changes are needed:
1st: Replace `neotraverse/legacy` with `neotraverse/dist/legacy/legacy.cjs`. I am not sure why node isn't picking up the regular import, everywhere else it worked.

2nd I assume has to do with `get`, but I'd like to inspect some more.

Reply:
I'll give that a try.

Reply:
I figured out the import issue, v0.6.12 fixes that. Could you bump it up? @wojtekmaj